src/sentry/utils/sdk.py

```diff
@@ -299,6 +328,14 @@ def capture_event(self, event):
             self._capture_anything("capture_event", event)
 
         def _capture_anything(self, method_name, *args, **kwargs):
+            # Experimental events will be sent to the experimental transport.
+            if experimental_transport:
+                rate = options.get("store.use-experimental-dsn-sample-rate")
+                if is_current_event_experimental():
+                    if rate and random.random() < rate:
+                        getattr(experimental_transport, method_name)(*args, **kwargs)
+                    # Experimental events should not be sent to other transports even if they are not sampled.
```

Comment:
I don't think you can do that. By setting the experimental flag on the scope you're effectively dropping the transaction for that web request. Or do you intend to push a new scope for producing the experimental event?

Reply:
It's pushing a new scope for capturing the exception and setting the tag there, so only the error event will be sent to the experimental dsn.

Reply:
https://github.com/getsentry/sentry-python/commit/caa41400423fc47bfac1d373b15304cfb0a7d8b5#diff-771cb012ea65fcb826b77e2abb36b504fdf3407fbe8054857b05fe5a41ed5164R595-R599

Reply:
ok. if this scope leaks for some reason (concurrency bug of some sort) we're royally screwed.

may i suggest creating a completely separate SDK setup (not even using the same client)?

```python
basic_client = Hub(Client(mydsn))


with basic_client:
    capture_event(...)
```

inside of the with-statement, you are using the new client and can send your events. the events are on a separate transport queue.

the scope is not the same, all request data will be missing from your events. but on the flipside this is much stronger isolation

this is how pytest-sentry isolates things

Reply:
> scope leaks for some reason

Like how the transports were leaking before? 😆 

I don't really expect the scope to leak, do we have any active issues with scope leaking that would make this more of a concern? The failure case, (being screwed here, lol) would only happen flipping the rate switch for the experimental dsn, which we can immediately turn off, and just for SaaS events, I feel like a separate client might be a bit too defensive?